src/v/raft/consensus.cc

```diff
@@ -1944,6 +1944,21 @@ consensus::next_followers_request_seq() {
     return ret;
 }
 
+ss::future<> consensus::refresh_commit_index() {
```

Comment:
I know that it would be more effort but maybe it would be better to allow consensus to apply stm batches up to the `high_watermark` instead. This way we would maintain the same guarantees as batches replicated by raft group. If there is no time to do it know maybe we can create a follow up ticket to handle this in future ?

Reply:
Either I don't understand your idea or it's too much work. Can you write pros/cons of it? So far we apply a command to `state_machine` only if it was fully replicated (same guarantees as acks=-1). 

`partition::high_watermark` is based on `consensus::last_visible_index` which a function of `_majority_replicated_index` which is the majority of the dirty offsets. So in theory `high_watermark` may be rolled back during a coordinated disk failure. But `state_machine` doesn't undo the commands so if we want to apply up to `high_watermark` then we need to redesign `state_machine` API too.

Reply:
You are right about this we would have to change the stm implementation. After giving it another thought I think it is correct. The STM can ask raft to flush. I am wondering about latency implications of this approach.

Reply:
It's an infrequent operation which happens when we delete the log's prefix (deletion clean up policy) so the latency impact should be eclipsed by that op.